extreme_classification/hierarchicalXC/classifier.py

```diff
@@ -89,10 +83,11 @@ def predict(self, X):
         X = X.toarray()
         start_id = len(self.merge_iterations) - 1
         classes = ssp.lil_matrix(np.zeros((len(X), self.num_classes)))
-        self.traverse_classifiers(start_id, X, classes)
+        ids = np.arange(len(X))
+        self.traverse_classifiers(start_id, X, ids, classes)
         return classes
 
-    def traverse_classifiers(self, current_id, X, classes):
+    def traverse_classifiers(self, current_id, X, this_ids, classes):
```

Comment:
wow, we were doing this?

Reply:
Doing what?

Reply:
Passing data redundantly.

Reply:
No. A slice was being passed.

Reply:
But I had this written it this way at some point, but got confused in between and reverted. Now fixed it again.

Reply:
cool

Reply:
Actually, will passing the whole thing create a copy? If I understand correctly, the same object will be passed around right?

Reply:
I am not sure. What you can do is make `X` a member of the class, and not pass it around to prevent copies from being created.

Reply:
Done